src/hooks/use-categories-names.tsx

```diff
@@ -1,22 +1,27 @@
 import { useCallback } from 'react'
 import { defaultResponses } from '~/constants'
 
-import useAxios from '~/hooks/use-axios'
+import useQuery from '~/hooks/use-query'
 import { categoryService } from '~/services/category-service'
-import { CategoryNameInterface } from '~/types'
 
 const useCategoriesNames = ({ fetchOnMount = true } = {}) => {
   const getCategoriesNames = useCallback(
     () => categoryService.getCategoriesNames(),
     []
   )
 
-  const { loading, response, fetchData, error } = useAxios<
-    CategoryNameInterface[]
-  >({
-    service: getCategoriesNames,
-    fetchOnMount,
-    defaultResponse: defaultResponses.array
+  const {
+    isLoading: loading,
+    data: response,
+    refetch: fetchData,
+    error
+  } = useQuery({
+    queryKey: ['categories-names'],
+    queryFn: getCategoriesNames,
```

Comment:
You don't need additional callback
```suggestion
    queryFn: categoryService.getCategoriesNames,
```

Reply:
I agree, thanks.